arm-network/2016-09-01/swagger/routeFilter.json

```diff
@@ -0,0 +1,849 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of ServiceCommunity resources.",
+            "schema": {
+              "$ref": "#/definitions/ServiceCommunityListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RouteFilterRulePropertiesFormat": {
+      "properties": {
+        "access": {
+          "type": "string",
+          "description": "The access type of the rule. Possible values are: 'Allow', 'Deny",
+          "enum": [
+            "Allow",
+            "Deny"
+          ],
+          "x-ms-enum": {
+            "name": "Access",
+            "modelAsString": true
+          }
+        },
+        "routeFilterRuleType": {
+          "type": "string",
+          "description": "The type of the rule. Possible values are: 'Community",
+          "enum": [
+            "Community"
+          ],
+          "x-ms-enum": {
+            "name": "RouteFilterRuleType",
+            "modelAsString": true
+          }
+        },
+        "communities": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The collection for bgp communities to filter on."
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "The provisioning state of the resource. Possible values are: 'Updating', 'Deleting', 'Succeeded' and 'Failed'."
+        }
+      },
+      "required": [
+        "nextHopType"
+      ],
+      "description": "Route Filter Rule Resource"
+    },
+    "RouteFitlerRule": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RouteFilterRulePropertiesFormat"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Route Filter Rule Resource"
+    },
+    "RouteFilterPropertiesFormat": {
+      "properties": {
+        "rules": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RouteFitlerRule"
```

Comment:
This is a recursive model. Calling this out as this looks strange. Wonder how flattening would look like in this scenario.